controllers/api/base_api_controller.js

```diff
@@ -15,56 +15,135 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-module.exports = function(pb) {
+module.exports = function BaseApiControllerModule(pb) {
+
+  // PB dependencies
+  const util = pb.util;
+  const BaseObjectService = pb.BaseObjectService;
+
+  /**
+   *
+   * @class BaseApiController
+   * @constructor
+   */
+  class BaseApiController extends pb.BaseController {
+
+    constructor() {
+      super();
+      /**
+       * Processes the query string to develop the where clause for the query request
+       * @method processWhere
+       * @param {Object} q The hash of all query parameters from the request
+       * @return {Object}
+       */
+      this.processWhere = function processWhere() {
```

Comment:
We would prefer to keep the constructor simple and clean.  Declaring the instance functions in the constructor nests them deeper than necessary and makes testing harder because you can't stub the prototype before the instance is created.